src/persist-client/src/internal/compact.rs

```diff
@@ -185,67 +193,80 @@ impl Compactor {
     {
         let parts_cpu_heavy_runtime = Arc::clone(&cpu_heavy_runtime);
         let compact_blocking = async move {
-            let () = Self::validate_req(&req)?;
-
-            let mut parts = BatchParts::new(
-                cfg.batch_builder_max_outstanding_parts,
-                Arc::clone(&metrics),
-                req.shard_id,
-                writer_id,
-                req.desc.lower().clone(),
-                Arc::clone(&blob),
-                parts_cpu_heavy_runtime,
-                &metrics.compaction.batch,
-            );
+            let () = Compactor::validate_req(&req)?;
+            // compaction needs memory enough for at least 2 fetched parts + 1 in-progress part
```

Comment:
last time I thought this over, I convinced myself we actually need 2 for in-progress because there's a moment where we have it in-mem both as a vec and as encoded parquet

Reply:
agreed! this was another I snuck past you while reviewing, all the comments / numbers are revised to account for this